__tests__/integration/create-nft.test.js

```diff
@@ -0,0 +1,183 @@
+    done();
+  });
+
+  it('should reject for a change address outside of the wallet', async done => {
+    const nftErr = await wallet1.createNft({
+      ...nftData,
+      amount: 10,
+      change_address: await wallet2.getAddressAt(0),
+      dontLogErrors: true
+    })
+      .catch(err => err);
+
+    expect(nftErr).toBeInstanceOf(Error);
+    expect(nftErr.response.text).toContain('Change address');
+    done();
+  });
+
+  it('should create nft with only mandatory parameters', async done => {
+    const htrBalanceBefore = await wallet1.getBalance();
+    const nextAddress = await wallet1.getNextAddress();
+
+    const nftTx = await wallet1.createNft({
+      ...nftData,
+      amount: 1
+    });
+    expect(nftTx.hash).toBeDefined();
+    // No authority tokens
+    for (const output of nftTx.outputs) {
+      expect(
+        output.token_data === TOKEN_DATA.HTR
+        || output.token_data === TOKEN_DATA.TOKEN
+      ).toBe(true);
+    }
+
+    // The fees (1) and deposits (1) should be deducted
+    const htrBalanceAfter = await wallet1.getBalance();
+    expect(htrBalanceAfter.available).toBe(htrBalanceBefore.available - 2);
+
+    // NFT should be on the next available address
+    const addrInfo = await TestUtils.getAddressInfo(nextAddress, wallet1.walletId, nftTx.hash);
+    expect(addrInfo.total_amount_available).toBe(1);
+
+    done();
+  });
+
+  it('should create nft with address and change address', async done => {
+    const destAddr = await wallet1.getNextAddress(true);
+    const changeAddr = await wallet1.getNextAddress(true);
+
+    const nftTx = await wallet1.createNft({
+      ...nftData,
+      amount: 1,
+      address: destAddr,
+      change_address: changeAddr
+    });
+    expect(nftTx.hash).toBeDefined();
+
+    // No authority tokens
+    let changeAmount;
+    for (const output of nftTx.outputs) {
+      expect(
+        output.token_data === TOKEN_DATA.HTR
+        || output.token_data === TOKEN_DATA.TOKEN
+      ).toBe(true);
+
+      // Fetching the change amount from the tx outputs ( htr output that is not the fee )
+      if (output.token_data === TOKEN_DATA.HTR && output.value !== 1) {
+        changeAmount = output.value;
+      }
+    }
+
+    // Validating balances for target addresses
+    const destInfo = await TestUtils.getAddressInfo(destAddr, wallet1.walletId, nftTx.hash);
+    expect(destInfo.total_amount_available).toBe(1);
+    const changeInfo = await TestUtils.getAddressInfo(changeAddr, wallet1.walletId);
+    expect(changeInfo.total_amount_available).toBe(changeAmount);
+
+    done();
+  });
+
+  it('should create nft with mint authority', async done => {
+    const nftTx = await wallet1.createNft({
+      ...nftData,
+      amount: 1,
+      create_mint: true
+    });
+
+    expect(nftTx.hash).toBeDefined();
+
+    // Validating authority tokens
+    const authorityOutputs = nftTx.outputs.filter(o => o.token_data === TOKEN_DATA.AUTHORITY_TOKEN);
```

Comment:
Great point! This would give a headache for a future test developer.
Fixed this on [a1431c7](https://github.com/HathorNetwork/hathor-wallet-headless/pull/164/commits/a1431c77636db8cb76c4aa78a08e33762bdbd0fd)